File: python/elasticbeanstalk/elasticbeanstalk-environment/elasticbeanstalk_environment/elasticbeanstalk_environment_stack.py
from aws_cdk import core
from aws_cdk.aws_elasticbeanstalk import (
    CfnApplication,
    CfnEnvironment
)


class ElasticbeanstalkEnvironmentStack(core.Stack):

    def __init__(self, scope: core.Construct, id: str, **kwargs) -> None:
        super().__init__(scope, id, **kwargs)

        node = self.node

        app_name = 'MyApp'

        platform = node.try_get_context("platform")

        app = CfnApplication(
            self, 'Application',
            application_name=app_name
        )

        CfnEnvironment(
            self, 'Environment',
            environment_name='MySampleEnvironment',
            application_name=app.application_name or app_name,
            platform_arn=platform
        )

File: csharp/elasticbeanstalk/elasticbeanstalk-environment/src/ElasticbeanstalkEnvironment/ElasticbeanstalkEnvironmentStack.cs
using Amazon.CDK;
using Amazon.CDK.AWS.ElasticBeanstalk;
using Constructs;

namespace ElasticbeanstalkEnvironment
{
    public class ElasticbeanstalkEnvironmentStack : Stack
    {
        public ElasticbeanstalkEnvironmentStack(Construct scope, string id, IStackProps props = null) : base(scope, id)
        {
            const string appName = "MyApp";

            var platform = this.Node.TryGetContext("platform").ToString();

            var app = new CfnApplication(this, "Application", new CfnApplicationProps
            {
                ApplicationName = appName
            });

            new CfnEnvironment(this, "Environment", new CfnEnvironmentProps
            {
                EnvironmentName = "MySampleEnvironment",
                ApplicationName = app.ApplicationName,
                PlatformArn = platform
            });
        }
    }
}
